lua/swenv/create.lua

```diff
@@ -0,0 +1,189 @@
+    python_app = 'python'
+  end
+  Job:new({
+    command = python_app,
+    args = { '-m', 'venv', venv_path },
+    cwd = dir_name,
+    on_exit = function(j, _)
+      vim.schedule(function()
+        if j.code ~= 0 then
+          vim.notify('swenv.nvim: ' .. vim.inspect(j._stderr_results .. j._stdout_results), vim.log.levels.ERROR)
+        else
+          local pip_path = venv_path .. '/' .. 'bin/pip'
+          local install_args = { 'install', '-r', requirements_path }
+          if requirements_path == 'pyproject.toml' then
+            install_args = { 'install', '.' }
+          end
+          Job:new({
+            command = pip_path,
+            args = install_args,
+            cwd = dir_name,
+            on_exit = function(k, _)
+              vim.schedule(function()
+                if k.code ~= 0 then
+                  vim.notify('swenv.nvim: ' .. vim.inspect(k._stderr_results), vim.log.levels.ERROR)
+                else
+
+                  local venv_name = vim.fs.basename(dir_name)
+                  swenv_set_venv(venv_path, venv_name)
+                  vim.notify('Set venv: ' .. venv_path, vim.log.levels.INFO)
+                end
+              end)
+            end,
+          }):start()
+        end
+      end)
+    end,
+  }):start()
+end
+
+--- Automatically create venv directory and use multiple method to auto install dependencies
+M.auto_create_set_python_venv = function()
+  local stop = false
+  local check_paths = {
+    {
+      path = venv_dir,
+      callback = function(path)
+        -- initial set, still want to do dependency install from others if available
+        local venv_name = vim.fs.basename(vim.fs.dirname(path))
+        swenv_set_venv(path, venv_name)
+      end,
+    },
+    {
+      path = 'pdm.lock',
+      callback = function(path)
+        pdm_sync(path)
+        vim.g._auto_set_python_venv_parent_dir = vim.fs.dirname(path)
+        stop = true
+      end,
+    },
+    {
+      path = 'requirements.txt',
+      callback = function(path)
+        pip_install_with_venv(path)
+        vim.g._auto_set_python_venv_parent_dir = vim.fs.dirname(path)
+        stop = true
+      end,
+    },
+    {
+      path = 'dev-requirements.txt',
+      callback = function(path)
+        pip_install_with_venv(path)
+        vim.g._auto_set_python_venv_parent_dir = vim.fs.dirname(path)
+        stop = true
+      end,
+    },
+    {
+      path = 'pyproject.toml',
+      callback = function(path)
+        pip_install_with_venv(path)
+        vim.g._auto_set_python_venv_parent_dir = vim.fs.dirname(path)
+        stop = true
+      end,
+    },
+  }
+
+  for _, val in ipairs(check_paths) do
+    local found_path = nil
+    local search_path = val['path']
+    local callback = val['callback']
+    if stop then
+      return
+    end
+    found_path = search_up(search_path)
+    if found_path ~= nil then
+      local last_parent_dir = vim.g._auto_set_python_venv_parent_dir
```

Comment:
does this need to be stored as a vim global variable or could it be local to this module?